### PR TITLE
Fix CosmosDB for latest API version

### PR DESCRIPTION
**Pull request by berndverst**
# Description

CosmosDB changed how they indicate when an entity is not found.

### CODE REVIEW

File: state/azure/cosmosdb/cosmosdb.go [hunk 1]
```diff
@@ -685,14 +683,16 @@ func populatePartitionMetadata(key string, requestMetadata map[string]string) st
 }
 
 func isNotFoundError(err error) bool {
+	fmt.Printf("Check isNotFoundError: %v\n", err)
```
Comment (yaron2) [inline state/azure/cosmosdb/cosmosdb.go:686]: Why fmt.Println here?

**Comment by yaron2** [comment]
@berndverst was this merged by mistake?

File: state/azure/cosmosdb/cosmosdb.go [hunk 2]
```diff
@@ -685,14 +683,16 @@ func populatePartitionMetadata(key string, requestMetadata map[string]string) st
...
 	if err == nil {
 		return false
 	}
 
 	if requestError, ok := err.(*azcore.ResponseError); ok {
-		if requestError.ErrorCode == statusNotFound {
+		if requestError.StatusCode == 404 {
```
Comment (yaron2) [inline state/azure/cosmosdb/cosmosdb.go:692]: Linter calls this line out as well to replace 404 with http.StatusNotFound.

It's better to wait for a review and approval..

**Comment by ItalyPaleAle** [comment]
@berndverst this LGTM but please make a follow-up PR to fix the linter issues

**Pull request merged**